### PR TITLE
fix(Cache.php) handle real object cache on expired transient deletion

### PR DESCRIPTION
**Pull request by lucatume**

### CODE REVIEW

**Comment by lucatume** [comment]
Closing as this is not the correct fix.

File: src/Tribe/Cache.php [hunk 1]
```diff
@@ -13,6 +13,7 @@ class Tribe__Cache implements ArrayAccess {
 	const SCHEDULED_EVENT_DELETE_TRANSIENT = 'tribe_schedule_transient_purge';
 	const NO_EXPIRATION  = 0;
 	const NON_PERSISTENT = - 1;
+	const PERSISTENT_GROUP = 'tribe-events';
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :warning: Equals sign not aligned with surrounding assignments; expected 17 spaces but found 1 space

<sub>hash: fbc6638102e7ca25ff4df7a986949812</sub>

File: src/Tribe/Cache.php [hunk 2]
```diff
@@ -30,6 +31,7 @@ public function hook() {
 		}
 
 		add_action( self::SCHEDULED_EVENT_DELETE_TRANSIENT, [ $this, 'delete_expired_transients' ] );
+		add_action( 'tribe_cache_flush_expired_transients', 'wp_cache_flush' );
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: e40acc485779eee01c24c9fa72edd718</sub>

File: src/Tribe/Cache.php [hunk 3]
```diff
@@ -68,7 +70,7 @@ public function set( $id, $value, $expiration = 0, $expiration_trigger = '' ) {
 			// Add so we know what group to use in the future.
 			$this->non_persistent_keys[] = $key;
 		} else {
-			$group = 'tribe-events';
+			$group = self::PERSISTENT_GROUP;
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 8e0781f0c9def131ab6b3b7c318d27d7</sub>

File: src/Tribe/Cache.php [hunk 4]
```diff
@@ -100,7 +102,7 @@ public function set_transient( $id, $value, $expiration = 0, $expiration_trigger
 	 * @return mixed
 	 */
 	public function get( $id, $expiration_trigger = '', $default = false, $expiration = 0, $args = array() ) {
-		$group = in_array( $id, $this->non_persistent_keys ) ? 'tribe-events-non-persistent' : 'tribe-events';
+		$group = in_array( $id, $this->non_persistent_keys ) ? 'tribe-events-non-persistent' : self::PERSISTENT_GROUP;
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: Line exceeds maximum limit of 100 characters; contains 118 characters

<sub>hash: 249f054ff4b5a5edc7b27f7cc8e5cc2c</sub>

File: src/Tribe/Cache.php [hunk 5]
```diff
@@ -141,7 +143,7 @@ public function get_transient( $id, $expiration_trigger = '' ) {
 	 * @return bool
 	 */
 	public function delete( $id, $expiration_trigger = '' ) {
-		return wp_cache_delete( $this->get_id( $id, $expiration_trigger ), 'tribe-events' );
+		return wp_cache_delete( $this->get_id( $id, $expiration_trigger ), self::PERSISTENT_GROUP );
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :warning: Line exceeds 80 characters; contains 100 characters

<sub>hash: 2f98f6f386dad0438a8ede62a6053aba</sub>

File: src/Tribe/Cache.php [hunk 6]
```diff
@@ -165,8 +167,7 @@ public function delete_expired_transients() {
 		global $wpdb;
 
 		$time = time();
-
-		$sql = "
+		$sql  = "
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 07cc09fcdf9df8ea1bca9bc575792a1e</sub>

File: src/Tribe/Cache.php [hunk 7]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
 				AND a.option_name NOT LIKE '\_transient\_timeout_tribe\_%'
 		";
 		$wpdb->query( $sql );
+
+		if ( wp_using_ext_object_cache() ) {
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 336ac3bf84d6723adbb3c9e45b213ac1</sub>

File: src/Tribe/Cache.php [hunk 8]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			/**
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 3f378d445a32c0f030d0065c13b32196</sub>

File: src/Tribe/Cache.php [hunk 9]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 * Toggle trigger to control if WordPress cache should be flushed or not.
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :warning: Line exceeds 80 characters; contains 85 characters

<sub>hash: c06240f09de27349c71c5dd52016a28c</sub>

File: src/Tribe/Cache.php [hunk 10]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 *
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: b58aaf574caa24018ec1b3ac517806a2</sub>

File: src/Tribe/Cache.php [hunk 11]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 * When using real object cache we cannot selectively flush a specific cache group.
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :warning: Line exceeds 80 characters; contains 95 characters

<sub>hash: d4d895f2e71eb4a97d90763c1d8e9648</sub>

File: src/Tribe/Cache.php [hunk 12]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 * This filter allows third-party code to handle the flush request in a more precise way by flushing only
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: Line exceeds maximum limit of 100 characters; contains 117 characters

<sub>hash: 3a8a0812848263852d2e4150a5c2916c</sub>

File: src/Tribe/Cache.php [hunk 13]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 * the persistent cache group.
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 6f32548982e7133864f9a87fee7903f4</sub>

File: src/Tribe/Cache.php [hunk 14]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 *
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: e478f4adf9eaf90719fc82ba1896a18f</sub>

File: src/Tribe/Cache.php [hunk 15]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 * @since TBD
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: There must be a single blank line after a tag group

<sub>hash: 58bcb85e83b3bce0d7ce7f2b54ce0acd</sub>

File: src/Tribe/Cache.php [hunk 16]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 *
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 3a15bf0cc9ef382a5b09a60f837974ed</sub>

File: src/Tribe/Cache.php [hunk 17]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 *
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 015ebd47871fd6577f1c00064f233b94</sub>

File: src/Tribe/Cache.php [hunk 18]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 * @param bool   $flush Whether to flush the WordPress cache or not.
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 5e70c8c22095799d1a06cdb89537d329</sub>

File: src/Tribe/Cache.php [hunk 19]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 * @param string $group The name of the group, `tribe-events`, we want to flush.
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :warning: Line exceeds 80 characters; contains 92 characters

<sub>hash: 2269ff59af6bdd61cd1051916161ac88</sub>

File: src/Tribe/Cache.php [hunk 20]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 */
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 0336cb59f27bbf3362fb33b89c204125</sub>

File: src/Tribe/Cache.php [hunk 21]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			$flush = apply_filters( 'tribe_cache_should_flush_expired_transients', true, self::PERSISTENT_GROUP );
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: TRUE, FALSE and NULL must be uppercase; expected "TRUE" but found "true"
* :no_entry_sign: Line exceeds maximum limit of 100 characters; contains 114 characters

<sub>hash: cf689214c6e497ca31c2bd5569ca8fbe</sub>

File: src/Tribe/Cache.php [hunk 22]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+
+			if ( ! $flush ) {
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 46d05cddb52196ae79e9166bdc5d35ff</sub>

File: src/Tribe/Cache.php [hunk 23]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+				return;
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: c07ad1a213394697d966453d3b98367d</sub>

File: src/Tribe/Cache.php [hunk 24]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			}
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: a3b73b6efa2be39d12c262eba5648802</sub>

File: src/Tribe/Cache.php [hunk 25]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+
+			/**
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 2512d75fedd94ff07fd54243156f6987</sub>

File: src/Tribe/Cache.php [hunk 26]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 * Fires when we need to delete expired transients in a WordPress installation using real object caching.
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: Line exceeds maximum limit of 100 characters; contains 117 characters

<sub>hash: a62eadc7f624a827626f55897435ae12</sub>

File: src/Tribe/Cache.php [hunk 27]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 *
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 5b7255395904ef29599e3dd322e355e8</sub>

File: src/Tribe/Cache.php [hunk 28]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 * @since TBD
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 4c6cc8316da2584d9519bcc4a01b212b</sub>

File: src/Tribe/Cache.php [hunk 29]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 *
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 16cf668f2fe1d8ff97bea258ad5674d9</sub>

File: src/Tribe/Cache.php [hunk 30]
```diff
@@ -180,6 +181,36 @@ public function delete_expired_transients() {
...
+			 * @param string $group The name of the group, `tribe-events`, we want to flush.
```
Comment (tr1b0t) [inline src/Tribe/Cache.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :warning: Line exceeds 80 characters; contains 92 characters

<sub>hash: 0b09302ab14c80a87679e3fb6f20eddc</sub>

**Pull request closed without merging**